_posts/2020-04-20-wgpu-web.md

```diff
@@ -0,0 +1,67 @@
+
+![Boids](/img/wgpu-boids.png)
+
+In the above screenshot, the same example is running inside of Firefox Nightly (top left), Chrome Canary (bottom left), and natively (right). [This example](https://github.com/gfx-rs/wgpu-rs/tree/master/examples/boids) uses a compute shader to [simulate flocking](https://en.wikipedia.org/wiki/Boids) and renders the result. macOS is used in the screenshot, but this example also runs on Windows and Linux without any code changes.
+
+All `wgpu-rs` examples are currently available at <http://joshgroves.com/gfx/wgpu/examples/>. [The WebGPU API has to be enabled in the browser](https://github.com/gfx-rs/wgpu-rs/pull/193#issuecomment-613182772) in order to run these examples.
+
+## Getting started with wgpu-rs on the web
+
+To take advantage of the new web backend in `wgpu-rs` and run on the web, crates can be compiled using the `wasm32-unknown-unknown` target. Because the WebGPU API is experimental, for now an unstable APIs flag (`--cfg=web_sys_unstable_apis`) must be provided in the `RUSTFLAGS` environment variable in order to use WebGPU from `web-sys` (the crate which exposes web APIs to Rust).
+
+After the crate has been compiled with `wasm32-unknown-unknown`, the WebAssembly can be executed within a browser. [`wasm-bindgen`](https://github.com/rustwasm/wasm-bindgen) can be used to generate JavaScript bindings, or the WebAssembly can be called manually.
+
+For example, to compile the [`hello-triangle`](https://github.com/gfx-rs/wgpu-rs/tree/master/examples/hello-triangle) example and generate bindings with `wasm-bindgen`:
+
+```bash
+# Build with the wasm target
+RUSTFLAGS=--cfg=web_sys_unstable_apis cargo build --target wasm32-unknown-unknown --example hello-triangle
+# Generate bindings with wasm-bindgen-cli into a `generated` directory
+cargo install -f wasm-bindgen-cli && wasm-bindgen --out-dir generated --web target/wasm32-unknown-unknown/debug/examples/hello-triangle.wasm
+```
+
+Then add a simple `index.html` inside the `generated` directory to run the WebAssembly:
+
+```html
+<html>
+  <body>
+    <script type="module">
+      import init from "./hello-triangle.js";
+      init();
+    </script>
+  </body>
+</html>
+```
+
+Now run a web server from the `generated` directory to see `hello-triangle` in the browser:
+
+![Hello Triangle](/img/wgpu-hello-triangle.png)
+
+We are really excited to see `wgpu-rs` projects running on the web. Even though it's still early days for WebGPU, we look forward to the new possibilities in high-performance, portable graphics and compute that this will enable. Please let us know if your project takes advantage of this new web backend.
```

Comment:
Maybe expand on how the reader is able to contact us and include links to our Matrix, Repo(s) etc.